### ISSUE TITLE
Support for font size scaling

### ISSUE THREAD

**Issue by brunoabinader** [mapbox/mapbox-gl-native#2429]
Mapbox API should have support for font size scaling. This enables applications to better adapt their interfaces for people with visual impairment.

Font size scaling information comes from client side. I propose a `fontScale` getter/setter in `mbgl::Map` class that defaults to `1.0` (float). This value acts as a multiplier when rendering fonts, with higher values increasing the font sizes specified in the style, and lower values decreasing accordingly. 

Thoughts?

/cc @mapbox/gl @mapbox/mobile

**Comment by kkaefer** [comment]
I agree; iOS also has system-wide settings that allow increasing the font size, and developers should be able to forward that setting to Mapbox GL. I imagine Android has a similar setting.

**Comment by friedbunny** [comment]
Cool idea, on the platform level this would be simple to hook up.

**Comment by mikemorris** [comment]
:+1:

**Comment by ansis** [comment]
Should we scale everything instead of only text?

- it could be useful to have non-text things like roads be more visible too
- scaling only text could mess up proportions/density and make designing more complicated

@andreasviglakis @nickidlugash which do you think would be preferable from a design perspective?

**Comment by andreasviglakis** [comment]
would it have to be all or nothing? i can imagine scenarios where we'd only want fonts and sprites to be enlarged.

**Comment by tobrun** [comment]
>  I imagine Android has a similar setting.

Yes, we can use [font scale attribute] for that (http://developer.android.com/reference/android/provider/Settings.System.html#FONT_SCALE)

**Comment by ansis** [comment]
> would it have to be all or nothing? i can imagine scenarios where we'd only want fonts and sprites to be enlarged.

It's definitely possible to scale only text/icons

**Comment by nickidlugash** [comment]
I'm looking now at Apple Maps on iOS with the largest font option; they basically only scale up text, and it seems pretty good. The label density is very low of course, but I'm not sure it would be possible or even desirable to try to maintain too much density. I'm also not sure scaling up other features will get us much more accessibility than just zooming in/out and/or using the native zoom feature (which on iOS anyway seems to work pretty well even though it decreases resolution).

Apple also scales up all icons with text, e.g. highway shields, which would also be necessary for us to do. They don't scale up other icons, but, as @andreasviglakis mentioned, that might be useful too (and maybe even easier? Since our sprites assets are svgs and we can probably just output a larger sprite?)

If we need to dynamically adjust any other properties due to the increased font size, would that be an option? I think most (if not all) will be fine, but perhaps something like `symbol-spacing` might need to be adjusted.

**Comment by incanus** [comment]
One approach might be to start with OS accessibility settings affecting just text, then see how that plays, then even consider a GL style spec option for specifying other features that could vary this way?

**Comment by brunoabinader** [comment]
> One approach might be to start with OS accessibility settings affecting just text

:+1: for text as a start point. However, @andreasviglakis pointed out that some sprites (eg. _highway shields_) would need to be scaled as well.

**Comment by 1ec5** [comment]
#7030 is another take that would happen entirely at the SDK level (depending on some mbgl changes that have to happen anyways for other features).

**Comment by felixLam** [comment]
For anyone interested we have open-sourced a "simple" approach to adjusting the fonts to match the user's preferred content size: https://github.com/iosphere/ISHMapboxDynamicFontObserver
